src/components/cards/FiatExchangeDetailsCard.tsx

```diff
@@ -113,20 +114,17 @@ export function FiatExchangeDetailsCard(props: Props) {
     sourceCurrencyCode = currencyCode
     destinationCurrencyCode = fiatAsset.fiatCurrencyCode
   }
-  useAsyncEffect(
-    async () => {
-      const exchangeAmount = nativeAmount != null ? await wallet.nativeToDenomination(nativeAmount, currencyCode) : ''
-      if (direction === 'buy') {
-        setSourceAmount(`${fiatAsset.fiatAmount} `)
-        setDestinationAmount(`${exchangeAmount} `)
-      } else {
-        setDestinationAmount(`${fiatAsset.fiatAmount} `)
-        setSourceAmount(`${exchangeAmount} `)
-      }
-    },
-    [],
-    'FiatExchangeDetailsCard'
-  )
+  useMount(() => {
+    const multiplier = getCurrencyCodeMultiplier(wallet.currencyConfig, currencyCode)
+    const exchangeAmount = nativeAmount != null ? div(nativeAmount, multiplier, multiplier.length) : ''
+    if (direction === 'buy') {
+      setSourceAmount(`${fiatAsset.fiatAmount} `)
+      setDestinationAmount(`${exchangeAmount} `)
+    } else {
+      setDestinationAmount(`${fiatAsset.fiatAmount} `)
+      setSourceAmount(`${exchangeAmount} `)
+    }
+  })
```

Comment:
We don't even need the "useMount" hook nor the component state for sourceAmount/destinationAmount. Instead we can just define `sourceAmount` and `destinationAmount` as local variables.